storage/listkeys.go

```diff
@@ -5,14 +5,13 @@ package storage
 
 // ListKeys returns keys starting from first and upto limit
 func ListKeys(store KeyValueStore, first Key, limit Limit) (Keys, error) {
-	const unlimited = Limit(1 << 31)
-
-	keys := make(Keys, 0, limit)
-	if limit == 0 {
-		// TODO: this shouldn't be probably the case
-		limit = unlimited
+	if limit > LookupLimit {
+		return nil, ErrLimitExceeded
```

Comment:
Wouldn't it better to override the user-defined limit with the internal `LookupLimit`? This proposal is just for the list methods, not for GetAll.

Reply:
Not feeling in any particular direction.

Reply:
fixed